tests/unit/src/kakarot/precompiles/test_precompiles.py

```diff
@@ -0,0 +1,31 @@
+import random
+import re
+
+import pytest
+import pytest_asyncio
+from starkware.starknet.testing.contract import StarknetContract
+from starkware.starknet.testing.starknet import Starknet
+
+
+@pytest_asyncio.fixture(scope="module")
+async def precompiles(starknet: Starknet):
+    return await starknet.deploy(
+        source="./tests/unit/src/kakarot/precompiles/test_precompiles.cairo",
+        cairo_path=["src"],
+        disable_hint_validation=True,
+    )
+
+
+@pytest.mark.asyncio
+class TestPrecompiles:
+    async def test_precompiles_should_throw_on_not_implemented(self, precompiles):
+        # we choose an out of range precompile address to get the NotImplementedPrecompile error, and test if it is including the address in the error
+        # note: in our implementation, `Precompiles.is_precompile` checks if an address is within a given range before dispatching, so usually an out of range address would never be passed to `Precompiles.run`
+        address = 10
+        with pytest.raises(Exception) as e:
+            await precompiles.test__precompiles_should_throw_on_not_implemented(
+                address=address
+            ).call()
+
+        message = re.search(r"Error message: (.*)", e.value.message)[1]
+        assert message == "Kakarot: NotImplementedPrecompile " + str(address)
```

Comment:
if `LAST_PRECOMPILE_ADDRESS` is `0x9`, I'm quite surprised that `address = 10` raises this error

Reply:
I am calling `Precompile.runs` directly, which would normally get called after a `Precompiles.is_precompile`

Reply:
`run` calls `_exec_precompile` with `address = 10`, consequently `offset = 1 + 30` and I don't know where it jumps, but should not jump to a `not_implemented_precompile` right?

Reply:
I see what you mean! When the address is 10, we do see an error w/ message of `Kakarot: NotImplementedPrecompile 10` Am experimenting to to see what the behavior is from 10-100, for curiosity. Will edit back results.

Reply:
si in summary, we have:

```
Kakarot: NotImplementedPrecompile 0
don't raise
Kakarot: NotImplementedPrecompile 2
Kakarot: NotImplementedPrecompile 3
don't raise
Kakarot: NotImplementedPrecompile 5
don't raise
Kakarot: NotImplementedPrecompile 7
Kakarot: NotImplementedPrecompile 8
Kakarot: NotImplementedPrecompile 9
Kakarot: NotImplementedPrecompile 10
Kakarot: NotImplementedPrecompile 11
raise something else (unknown memory location?)
```

sounds weird!

Reply:
so bottom line is:
- the function `not_implemented_precompile` is defined right after the jump table
- when jumping to 10, one actually jump into `not_implemented_precompile`...
- when jumping to 11, one jumps somewhere in `not_implemented_precompile` and the program raises *but* error_message is set to `Kakarot: NotImplementedPrecompile 11`
- when jumping to 12 or higher, it does raise an invalid location error